instrumentation/nocode/src/main/java/com/splunk/opentelemetry/instrumentation/nocode/NocodeAttributesExtractor.java

```diff
@@ -38,10 +37,10 @@ public void onStart(
       AttributesBuilder attributesBuilder, Context context, NocodeMethodInvocation mi) {
     codeExtractor.onStart(attributesBuilder, context, mi.getClassAndMethod());
 
-    Map<String, String> attributes = mi.getRuleAttributes();
+    Map<String, NocodeExpression> attributes = mi.getRuleAttributes();
     for (String key : attributes.keySet()) {
-      String expression = attributes.get(key);
-      Object value = NocodeEvaluation.evaluate(expression, mi.getThiz(), mi.getParameters());
+      NocodeExpression expression = attributes.get(key);
```

Comment:
I don't feel super duper strongly about this, but anytime I see iteration over keyset where the first line of the loop is `map.get(key)` I feel like it just wants to be iterating over the `map.entrySet()` instead.